### PR TITLE
ARROW-2325: [Python] Update setup.py to use Markdown project description

### PR DESCRIPTION
**Pull request by n0mn0m**
Any thoughts on the best way to make sure the `pip` upload uses `twine>=1.11.0` and `setuptools>=38.6.0`? I don't see a reference to the twine version in https://github.com/apache/arrow/blob/master/dev/release/RELEASE_MANAGEMENT.md but maybe adding a note? Or maybe it should be documented and part of the packaging discussion?

### CODE REVIEW

**Review by wesm — APPROVED** [review]
+1. Feel free to document in RELEASE_MANAGEMENT.md

File: python/setup.py [hunk 1]
```diff
@@ -494,6 +493,7 @@ def parse_version(root):
     tests_require=['pytest', 'pandas'],
     description="Python library for Apache Arrow",
     long_description=long_description,
+    long_description_content_type="text/markdown",
```
Comment (pitrou) [inline python/setup.py]: Are unknown arguments simply ignored? I'm thinking of people who may want to build Arrow using an older setuptools.

File: python/setup.py [hunk 2]
```diff
@@ -39,6 +39,8 @@
 from distutils.util import strtobool
 from distutils import sysconfig
 
+from setuptools import setup
```
Comment (pitrou) [inline python/setup.py]: That import already exists above.

File: python/setup.py (hunk 1 above)
Comment (AlexHagerman) [reply python/setup.py]: I can create a new environment with `setuptools < 38.6.0` to see if building breaks.

File: python/setup.py (hunk 2 above)
Comment (AlexHagerman) [reply python/setup.py]: Not sure why I missed the existing and added this import. I'll get that corrected and also setup my linter to find duplicate imports. Thanks for catching.

File: python/setup.py (hunk 1 above)
Comment (AlexHagerman) [reply python/setup.py]: I get this warning: `UserWarning: Unknown distribution options: 'long_description_content_type' warnings.warn(msg)` but the build still completes when running `python setup.py build_ext --build-type=$ARROW_BUILD_TYPE --with-parquet --with-plasma --inplace` do you have any suggestions on a more formal or exhaustive way to check on this?

Comment (pitrou) [reply python/setup.py]: No, it's fine with a warning, as long as the build succeeds.

**Comment by AlexHagerman** [comment]
@pitrou I realize I've been using the flake8 commands from the development doc `flake8 pyarrow`
and `flake8 --config=.flake8.cython pyarrow`. When I run flake8 at the python folder level I see a few different lint errors, and the reimport of `setuptools` in `setup.py` prior to the removal. I will run flake8 at the python folder level in the future, but would it be worth opening a ticket or two about cleaning up the linter errors and then having the build run on the `python` folder not just `pyarrow`?
flake8_arrow

**Comment by xhochy** [comment]
@AlexHagerman We should also run `flake8` on these files but we can make that adjustment in a separate PR. We sadly cannot run it on the whole `arrow/python` directory as there will typically be other files like a developer's virtualenv.

**Comment by xhochy** [comment]
> Any thoughts on the best way to make sure the pip upload uses twine>=1.11.0 and setuptools>=38.6.0? I don't see a reference to the twine version in https://github.com/apache/arrow/blob/master/dev/release/RELEASE_MANAGEMENT.md but maybe adding a note? Or maybe it should be documented and part of the packaging discussion?

Adding a note to https://github.com/apache/arrow/blob/master/dev/release/RELEASE_MANAGEMENT.md should be enough for now. We need build better infrastructure for our release but this is sadly more mid-term than short-term.

**Comment by AlexHagerman** [comment]
@xhochy I went through the appveyor and travis files to get a better understanding before updating the release documentation. In arrow-dist/scripts/https://github.com/apache/arrow-dist/blob/master/scripts/python-wheels-windows.bat I see a new conda environment being created and `setuptools` is defaulting to 39.0.1 for new environments today, so we are good there. I was wondering if you could point me towards understanding when/how an environment is created with `.travis.yml`? Wanted to make sure `setuptools` would be up to date, or open a ticket if we needed to add something like `SETUPTOOLS_DEP>="38.6.0"`.

**Comment by xhochy** [comment]
@AlexHagerman We only use `conda` and `virtualenv` to create environments in Travis and AppVeyor. Both install the latest version of `setuptools` on the creation of new environments (`conda` is more explicit about that). So we're fine with both of them in the CI.

**Review by xhochy — APPROVED** [review]
+1, LGTM

**Pull request closed without merging**